index.js

```diff
@@ -121,6 +121,10 @@ module.exports = function (options) {
     throw new Error('No correct handler set. It\'s needed to actually process a Micropub request.');
   }
 
+  if (!options.syndication || options.syndication.constructor !== Array) {
```

Comment:
Yep agree with all this. A callback would be far more powerful.

I just went with defining the response as an array rather than a callback mostly because it's easier and I've not yet got my head around promises.